src/autocomplete.js

```diff
@@ -206,7 +207,7 @@ export default class Autocomplete extends Component {
 
     this.setState({ query })
 
-    const searchForOptions = !queryEmpty && queryChanged && queryLongEnough
+    const searchForOptions = showAllValues || (!queryEmpty && queryChanged && queryLongEnough)
```

Comment:
> Yeah I can see the logic however I think it's not intuitive to just have minlength 0 trigger quite different behaviour to the other min lengths.

I don't think the behaviour is different? But I agree that it is obfuscated.

I'm wary of providing conflicting options: `minLength: 3, showAllValues: true` doesn't make sense, because it shows all the options on first click and then hides them again until you get to 3 characters. Some autocompletes do implement a [alwaysRenderSuggestions](https://github.com/moroshko/react-autosuggest#alwaysRenderSuggestionsProp) option but they don't have something like our `minLength`.

I think it makes sense to merge behaviour here under the same option, but `minLength: 0` is weird.

How about replacing `minLength` with this: 

> #### `showValuesMinLength` (default: `1`)
> 
> Type: `number`
> 
> The minimum number of characters that should be entered before the autocomplete will attempt to suggest options. **If this is set to 0, the autocomplete will display suggestions when users first click or tab into it**. When the query length is under this, the aria status region will also provide helpful text to the user informing them they should type in more.

(added ** for emphasis)

We can alias and deprecate `minLength` to this then.